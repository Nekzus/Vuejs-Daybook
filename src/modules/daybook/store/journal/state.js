export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            text: 'Quis enim duis excepteur mollit occaecat est eu enim est incididunt elit ad.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            text: 'Excepteur cillum consectetur qui excepteur sunt ullamco voluptate do cillum tempor laboris pariatur dolore in.',
            picture: null,
        },
        {
            id: new Date().getTime() + 2000,
            date: new Date().toDateString(),
            text: 'Elit ex veniam tempor culpa duis labore ad qui Lorem eiusmod adipisicing duis.',
            picture: null,
        },
    ]
})