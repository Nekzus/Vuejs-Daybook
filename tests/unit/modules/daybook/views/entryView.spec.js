import { createStore } from 'vuex'
import { shallowMount } from '@vue/test-utils'
import Swal from 'sweetalert2'
import EntryView from '@/modules/daybook/views/EntryView'
import journal from '@/modules/daybook/store/journal'
import { journalState } from '../../../mock-data/test-journal-state'

const createVuexStore = initialState =>
    createStore({
        modules: {
            journal: {
                ...journal,
                state: { ...initialState },
            },
        },
    })

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn(),
}))

describe('Pruebas en el EntryView', () => {
    const store = createVuexStore(journalState)
    store.dispatch = jest.fn()

    const mockRouter = {
        push: jest.fn(),
    }

    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallowMount(EntryView, {
            props: {
                id: '-N2muSDCOVOu9WkgI0qT',
            },
            global: {
                mocks: {
                    $router: mockRouter,
                },
                plugins: [store],
            },
        })
    })

    test('debe de sacar al usuario porque el is no existe', () => {
        const wrapper = shallowMount(EntryView, {
            props: {
                id: 'Este ID no existe en el STORE',
            },
            global: {
                mocks: {
                    $router: mockRouter,
                },
                plugins: [store],
            },
        })
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })
    })

    test('debe de mostrar la entrada correctamente', () => {
        expect(wrapper.html()).toMatchSnapshot()
        expect(mockRouter.push).not.toHaveBeenCalled()
    })

    test('debe de borrar la entrada y salir', done => {
        Swal.fire.mockReturnValueOnce(Promise.resolve({ isConfirmed: true }))
        wrapper.find('.btn-danger').trigger('click')
        expect(Swal.fire).toHaveBeenCalledWith({
            title: '¿Estás seguro?',
            text: '¡No podrás revertir esto!',
            showDenyButton: true,
            confirmButtonText: 'Sí, borrar',
        })
        setTimeout(() => {
            expect(store.dispatch).toHaveBeenCalledWith(
                'journal/deleteEntry',
                '-N2muSDCOVOu9WkgI0qT',
            )
            expect(mockRouter.push).toHaveBeenCalled()
            done()
        }, 1)
    })
})
