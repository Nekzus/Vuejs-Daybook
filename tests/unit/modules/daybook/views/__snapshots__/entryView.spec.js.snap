// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView debe de mostrar la entrada correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3 fw-bold">23</span><span class="mx-1 fs-3">Mayo</span><span class="mx-2 fs-4 fw-light">2022, Lunes</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir foto <i class="fa fa-upload"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Qué sucedió hoy?"></textarea></div>
<img src="https://res.cloudinary.com/dsvsl0b0b/image/upload/v1653346811/kyz4dgbahmaimyyo1pp4.jpg" alt="entry-picture" class="img-thumbnail">
<!--v-if-->
<fab-stub icon="fa-save"></fab-stub>
`;
